addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py

```diff
@@ -275,6 +275,17 @@ def _import_fill_invoice_form(self, journal, tree, invoice_form, qty_factor):
                 logs.append(_("Could not retrieve currency: %s. Did you enable the multicurrency option and "
                               "activate the currency ?", currency_code_node.text))
 
+        # ==== Bank Details ====
+
+        bank_detail_nodes = tree.findall('.//{*}SpecifiedTradeSettlementPaymentMeans')
+        bank_details = [
+            bank_detail_node.findtext('{*}PayeePartyCreditorFinancialAccount/{*}IBANID')
+            or bank_detail_node.findtext('{*}PayeePartyCreditorFinancialAccount/{*}ProprietaryID')
+        for bank_detail_node in bank_detail_nodes]
```

Comment:
`findtext` either return the value found or None. In our case the IF condition on the `bank_detail_node` would not help since the said node always exists in the list creation since those nodes are found by the `findall` method. To avoid attempting to create a bank account with the acc_number=None (in the case `findtext` returns None), I have added the condition `and account_number:` in the `_import_retrieve_and_fill_partner_bank_details` method